### PR TITLE
Implement machine linking

### PR DESCRIPTION
**Pull request by ShadowCommander**
Transmitters need a TransmitterComponent and a component that triggers it.

Receivers need a ReceiverComponent and a component that implements IReceiver.

Implemented a Switch
Receiver functionality to WallLight

TODO:
- [x] Linker for mapper
- [x] Error for transmitters or receivers without links to help mappers

### CODE REVIEW

File: Content.Server/GameObjects/Components/MachineLinking/IReceiver.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using Robust.Shared.GameObjects;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Content.Server.GameObjects.Components.MachineLinking
+{
+    public interface IReceiver
```
Comment (Zumorica) [inline Content.Server/GameObjects/Components/MachineLinking/IReceiver.cs:8]: I wonder if we could make this an `IReceiver<T>` so triggering could take any argument, hmm.

**Comment by RemieRichards** [comment]
I was going to complain about Transmitter and Receiver being too broad, but you've already renamed them to Signal* so 👍

**Comment by ike709** [comment]
Resolves #1656

**Comment by ShadowCommander** [comment]
Ready for review

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**